### DISCUSSION TITLE
Navigation Link

### DISCUSSION THREAD

**Discussion by sachinsardana** [DominikButz/MinimizableView#8] [General]
I'm using this library and it works amazingly, but I'm getting a problem... When you use the navigation link on the music player view (presented view), the mini player is hidden, and again presenting the music player from the navigated screen app causes the app to crash with the error "Thread 1: Fatal error:No observable object of type MinimizeViewHandler was found. "A View.environmentObject(_:) for MinimizableViewHandler as an ancestor of this view may be missing."

How do I handle this?

**Comment by DominikButz** [comment]
Hi,
thanks for reaching out to me. After the minimizableView modifier, make sure to add this modifier:
    .environmentObject(self.miniHandler)
Also make sure to pass the miniHandler as environment object to all views where miniHandler is used for controlling minimized and presented state. 
regards
Dominik

**Reply by sachinsardana** [reply]
Not working,using below code
**Root Class**
@ObservedObject var miniHandler: MinimizableViewHandler = MinimizableViewHandler()
      .minimizableView(content: {DuplicateMusicPlayerViewC(animationNamespaceId: self.namespace)
                        .ignoresSafeArea()
                    },
                                     compactView: {
                        MiniMusicView()
                            .environmentObject(self.miniHandler)
                    },

**Navigated View**
 @EnvironmentObject var miniHandler: MinimizableViewHandler

**Comment by DominikButz** [comment]
That’s what I wrote, you are not passing the mini handler to the minimizableView modifier.

You are only passing it to your compact view!

Recheck the example project

Regards
Dominik

**Comment by sachinsardana** [comment]
<img width="439" alt="Screenshot 2023-01-30 at 4 10 49 PM" src="https://user-images.githubusercontent.com/56502185/215454929-94ecb3f2-5b93-4f95-af15-72a8e1ab7dc3.png">

Please check this screenshot,already following what you asked to do.

**Reply by sachinsardana** [reply]
Player never gets hide in my complete app untill i navigate to other screen from MusicPlayer view----On Navigating to other view from music player and presenting player again from navigated view... app crashes with reason-"Thread 1: Fatal error:No observable object of type MinimizeViewHandler was found. "A View.environmentObject(_:) for MinimizableViewHandler as an ancestor of this view may be missing."
Already using this on navigated screen
@EnvironmentObject var miniHandler: MinimizableViewHandler

**Comment by DominikButz** [comment]
I use miniView in an app and this type of crash never happened. I can't solve the problem by just looking at code snippets. moreover, a complete app is much too complicated to just describe it without any coding details.... I suggest you create a minimal reproducible example (MRE) and send the Xcode project to me. I need to see it hands on in the simulator (or on a real device) to be able to pinpoint the cause of the error.  
By the way, it is better to initialise an environment object using @StateObject, not @ObservedObject. ObservedObject is actually only used as parameter in a subview when you pass a viewModel from a parent view down to the subview..

**Reply by sachinsardana** [reply]
[MinimizableViewExample-master 2.zip](https://github.com/DominikButz/MinimizableView/files/10553890/MinimizableViewExample-master.2.zip)
Updated Some code for new screen navigation,getting crash here too

**Reply by sachinsardana** [reply]
Added **.environmentObject(self.miniHandler)**,after navigation link end----Now miniplayer is not visible to next screen but app is not getting crash on accessing minihandler object

**Reply by DominikButz** [reply]
I can't follow. You are just sending the example project to me that I created myself. Other than tab view navigation, there is no navigation. I launched it and it is working as expected. No crash.

**Reply by DominikButz** [reply]
now I got it... you added a view called NewView... the way it is set up is a bit weird. 


Try this: 
`
    if isGoToNextView{
        NavigationLink(destination: NewView().environmentObject(miniHandler),
                        isActive: $isGoToNextView){
            EmptyView()
        }
    }
`

**Reply by DominikButz** [reply]
By the way: I suggest not using Navigation link within mini view. I think that is the reason why mini view doesn't stay on top.  you should check out SwiftUI tutorials about programmatic navigation. E.g. set up a navigationViewModel which you can pass to your mini content view. Here is a tutorial about programmatic navigation with NavigationStack https://www.youtube.com/watch?v=oxp8Qqwr4AY

**Reply by sachinsardana** [reply]
okay thanks,will check this.
